File: ios/Classes/FlutterBluegpsSdkPlugin.swift
import Flutter
import UIKit

public class FlutterBluegpsSdkPlugin: NSObject, FlutterPlugin {
  private var config: [String: Any]?
  private var isPositioning: Bool = false
  
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "flutter_bluegps_sdk", binaryMessenger: registrar.messenger())
    let instance = FlutterBluegpsSdkPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("iOS " + UIDevice.current.systemVersion)
      
    case "initialize":
      guard let arguments = call.arguments as? [String: Any] else {
        result(FlutterError(code: "INVALID_CONFIG", 
                          message: "Configuration is required", 
                          details: nil))
        return
      }
      config = arguments
      // Initialize BlueGPS SDK here with config
      result(nil)
      
    case "startPositioning":
      guard config != nil else {
        result(FlutterError(code: "NOT_INITIALIZED", 
                          message: "SDK not initialized", 
                          details: nil))
        return
      }
      isPositioning = true
      // Start positioning logic here
      result(nil)
      
    case "stopPositioning":
      isPositioning = false
      // Stop positioning logic here
      result(nil)
      
    case "getCurrentPosition":
      guard isPositioning else {
        result(nil)
        return
      }
      // Get current position from BlueGPS SDK
      // For now, return a mock position
      let position: [String: Any] = [
        "x": 0.0,
        "y": 0.0,
        "floor": 0,
        "timestamp": Int64(Date().timeIntervalSince1970 * 1000),
        "accuracy": 1.0
      ]
      result(position)
      
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}

File: android/src/main/kotlin/com/synapseslab/flutter_bluegps_sdk/FlutterBluegpsSdkPlugin.kt
package com.synapseslab.flutter_bluegps_sdk

import androidx.annotation.NonNull
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

/** FlutterBluegpsSdkPlugin */
class FlutterBluegpsSdkPlugin: FlutterPlugin, MethodCallHandler {
  private lateinit var channel : MethodChannel
  private var config: Map<String, Any>? = null
  private var isPositioning: Boolean = false

  override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "flutter_bluegps_sdk")
    channel.setMethodCallHandler(this)
  }

  override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
    when (call.method) {
      "getPlatformVersion" -> {
        result.success("Android ${android.os.Build.VERSION.RELEASE}")
      }
      "initialize" -> {
        try {
          config = call.arguments as? Map<String, Any>
          if (config != null) {
            // Initialize BlueGPS SDK here with config
            result.success(null)
          } else {
            result.error("INVALID_CONFIG", "Configuration is required", null)
          }
        } catch (e: Exception) {
          result.error("INITIALIZATION_ERROR", e.message, null)
        }
      }
      "startPositioning" -> {
        try {
          if (config == null) {
            result.error("NOT_INITIALIZED", "SDK not initialized", null)
            return
          }
          isPositioning = true
          // Start positioning logic here
          result.success(null)
        } catch (e: Exception) {
          result.error("START_ERROR", e.message, null)
        }
      }
      "stopPositioning" -> {
        try {
          isPositioning = false
          // Stop positioning logic here
          result.success(null)
        } catch (e: Exception) {
          result.error("STOP_ERROR", e.message, null)
        }
      }
      "getCurrentPosition" -> {
        try {
          if (!isPositioning) {
            result.success(null)
            return
          }
          // Get current position from BlueGPS SDK
          // For now, return a mock position
          val position = mapOf(
            "x" to 0.0,
            "y" to 0.0,
            "floor" to 0,
            "timestamp" to System.currentTimeMillis(),
            "accuracy" to 1.0
          )
          result.success(position)
        } catch (e: Exception) {
          result.error("POSITION_ERROR", e.message, null)
        }
      }
      else -> {
        result.notImplemented()
      }
    }
  }

  override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
    channel.setMethodCallHandler(null)
  }
}
